crates/rpc-types-engine/src/forkchoice.rs

```diff
@@ -34,22 +34,25 @@ pub struct ForkchoiceState {
 ///
 /// These are considered hard RPC errors and are _not_ returned as [PayloadStatus] or
 /// [PayloadStatusEnum::Invalid].
-#[derive(Clone, Copy, Debug, PartialEq, Eq, thiserror::Error)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, derive_more::Display)]
```

Comment:
once we bump msrv to 1.81 we can also use derive_more error and remove the feature entirely

https://docs.rs/derive_more/latest/derive_more/derive.Error.html